doc/authoring_help.md

```diff
@@ -0,0 +1,95 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+#---------------------------------------------------------------------------------------------
+
+from azure.cli.help_files import helps
+
+#pylint: disable=line-too-long
+
+helps['account clear'] = """
+            type: command
+            short-summary: Clear account
+            long-summary: Longer summary of clearing account
+            parameters: 
+                - name: --account-name -n
+                  type: string
+                  short-summary: 'Account name'
+                  long-summary: |
+                      Longer summary with newlines preserved.  Preserving newlines is helpful for paragraph breaks.
+                  populator-commands: 
+                  - az account list
+                  - These indicate where values can be retrieved for input to this command
+                - name: --another-parameter
+                  short-summary: These parameter names must match what is shown in the command's CLI help output, including abbreviation.
+            examples:
+                - name: Document a parameter that doesn't exist
+                  text: >
+                    You will get an error when you show help for the command stating there is an extra parameter.
+                - name: Collapse whitespace in YAML
+                  text: >
+                    The > character collapses multiple lines into a single line, which is good for on-screen wrapping.
+            """
+</pre>
+
+You can also document groups using the same format.
+
+<pre>
+helps['account'] = """
+            type: group
+            short-summary: The account group
+            long-summary: Longer summary of account            
+            examples:
+                - name: Clear an account 
+                  text: Description
+                - name: Choose your current account
+                  text: az account set...
+            """
+</pre>
+
+# Testing Authored Help #
+
+To verify the YAML help is correctly formatted, the command/group's help command must be executed at runtime.  For example, to verify "az account clear", run the command "az account clear -h" and verify the text.  
+
+Runtime is also when help authoring errors will be reported, such as documenting a parameter that doesn't exist.  Errors will only show when the CLI help is executed, so verifying the CLI help is required to ensure your authoring is correct.   
+
+# Other Help Authoring #
+
+Commands without YAML usually still have help text.  Where does it come from?  These sections briefly outline where Az help text comes from.
+
+Authoring note: it is not recommended to use the product code to author command/group help--YAML is the recommended way (see above).  This information is provided for completeness and may be useful for fixing small typos in existing help text.
```

Comment:
for quick documentation, using the register help= format is fine.  If the whole/most of the command needs its help to be re-written, we're better off unifying the help into a single YAML string rather than registering each param just to add the help tag to it.  

---
In reply to: [73418361](https://github.com/Azure/azure-cli/pull/600#discussion_r73418361) [](ancestors = 73418361)